### PR TITLE
chore(CMSIS,PeriphDrivers): Sync latest MAX32657 changes

### PR DESCRIPTION
**Pull request by ozersa**
This commit content comes from feat/ME30 branch.
CMSIS register, includes files and peripheral source files copied from feat/ME30 branch to here to merge them into main.
These files requires to push MAX32657 development to the upstream.

### Checklist Before Requesting Review

- [X] PR Title follows correct guidelines.
- [ ] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

**Comment by ozersa** [comment]
@ttmut please check i3c_reva.c file, this file content not same for main and feat/ME30 branch,  the content  seems changed while merging  https://github.com/analogdevicesinc/msdk/pull/1111 
I keep main/i3c_reva.c file.

File: Libraries/PeriphDrivers/Include/MAX32657/i3c.h [hunk 1]
```diff
@@ -153,9 +153,9 @@ struct _i3c_ccc_req_t {
 };
 
 /**
- * @brief Private SDR/I2C request structure.
+ * @brief Private SDR request structure.
  *
- * SDR/I2C read and write to an I3C target.
+ * SDR read and write to an I3C target.
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/i3c.h:158]: Not critical but `SDR/I2C ...` is the correct comment.

**Comment by ttmut** [comment]
> @ttmut please check i3c_reva.c file, this file content not same for main and feat/ME30 branch, the content seems changed while merging #1111 I keep main/i3c_reva.c file.

`main/i3c_reva.c` is the correct one. Thanks.

File: Libraries/PeriphDrivers/Include/MAX32657/i3c.h (hunk 1 above)
Comment (ozersa) [reply Libraries/PeriphDrivers/Include/MAX32657/i3c.h:158]: Feel free to push a new commit to fix it.

**Review by Jake-Carter — COMMENTED** [review]
Thanks @ozersa 

I copied your merge conflict resolutions back onto #1075 

Let's just merge that one in now

**Comment by ozersa** [comment]
> Thanks @ozersa
> 
> I copied your merge conflict resolutions back onto #1075
> 
> Let's just merge that one in now

It is ok for me.

**Pull request closed without merging**